presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -46,11 +52,24 @@ public IterativeOptimizer(StatsRecorder stats, Set<Rule> rules)
 
     public IterativeOptimizer(StatsRecorder stats, List<PlanOptimizer> legacyRules, Set<Rule> newRules)
     {
+        this(stats, legacyRules, newRules, 5000);
```

Comment:
I thought I had made a comment about making the default timeout configurable and optional, but I can't find it now.

A timeout of 5s is not necessarily a good general setting. Some optimizers might need to access external metadata and could be sensitive to connector choice and other environmental properties.